kimchi/src/circuits/constraints.rs

```diff
@@ -146,6 +147,13 @@ pub struct ConstraintSystem<F: FftField> {
     #[serde_as(as = "o1_utils::serialization::SerdeAs")]
     pub endomul_scalar8: E<F, D<F>>,
 
+    /// Cairo constraint selector polynomials
+    #[serde_as(as = "[o1_utils::serialization::SerdeAs; turshi::CIRCUIT_GATE_COUNT]")]
+    pub cairo: [DP<F>; turshi::CIRCUIT_GATE_COUNT],
+    /// Cairo selector evaluations over domain.d8
+    #[serde_as(as = "[o1_utils::serialization::SerdeAs; turshi::CIRCUIT_GATE_COUNT]")]
+    pub cairo8: [E<F, D<F>>; turshi::CIRCUIT_GATE_COUNT],
```

Comment:
since Cairo is optional, I think these should live under some `Option<Cairo>` field

Reply:
amazing 😃